lib/money-rails/active_record/monetizable.rb

```diff
@@ -117,9 +117,19 @@ def monetize(*fields)
               memoized = instance_variable_get("@#{name}")
 
               # Dont create a new Money instance if the values haven't been changed.
-              if memoized && memoized.cents == amount &&
-                memoized.currency == attr_currency
-                result =  memoized
+              if memoized && memoized.cents == amount
+                if memoized.currency == attr_currency
+                  result =  memoized
+                else
+                  amount = memoized.amount
+                  unless amount.blank?
+                    amount = amount.to_money(attr_currency)
+                    write_attribute subunit_name, amount.cents
+                  end
+
+                  # Cache the value (it may be nil)
+                  result = instance_variable_set "@#{name}", amount
```

Comment:
No need to set instance variable if the amount is blank, it's going to be exactly the same